src/jaxsim/rbda/contacts/visco_elastic.py

```diff
@@ -0,0 +1,860 @@
+                    joint_forces=references.joint_force_references(model=model),
+                    link_forces=W_f̅_L + references.link_forces(model=model, data=data),
+                )
+            )
+
+        # Compute the system mixed acceleration, used to integrate the system position.
+        # It considers the average of the average contact forces computed with the
+        # exponential integrator.
+        with (
+            data.switch_velocity_representation(jaxsim.VelRepr.Mixed),
+            references.switch_velocity_representation(jaxsim.VelRepr.Mixed),
+        ):
+            BW_ν̇_pr2 = jnp.hstack(
+                js.ode.system_acceleration(
+                    model=model,
+                    data=data,
+                    joint_forces=references.joint_force_references(model=model),
+                    link_forces=LW_f̿_L + references.link_forces(model=model, data=data),
+                )
+            )
+
+        # Integrate the system velocity using the inertial-fixed acceleration.
+        W_ν_plus = W_ν_t0 + dt * W_ν̇_pr
+
+        # Integrate the system position using the mixed velocity.
+        q_plus = jnp.hstack(
+            [
+                # Note: here both ṗ and p̈ -> need mixed representation.
+                W_p_B_t0 + dt * W_ṗ_B_t0 + 0.5 * dt**2 * BW_ν̇_pr2[0:3],
+                jaxsim.math.Quaternion.integration(
+                    dt=dt,
+                    quaternion=W_Q_B_t0,
+                    omega=(W_ω_WB_t0 + 0.5 * dt * BW_ν̇_pr2[3:6]),
+                    omega_in_body_fixed=False,
+                ).squeeze(),
+                s_t0 + dt * ṡ_t0 + 0.5 * dt**2 * BW_ν̇_pr2[6:],
+            ]
+        )
+
+        # Create the data at the final time.
+        with data.editable(validate=True) as data_tf:
+            data_tf: js.data.JaxSimModelData
+            data_tf.time_ns = data.time_ns + (dt * 1e9).astype(data.time_ns.dtype)
+
+        data_tf = data_tf.reset_joint_positions(q_plus[7:])
+        data_tf = data_tf.reset_base_position(q_plus[0:3])
+        data_tf = data_tf.reset_base_quaternion(q_plus[3:7])
+
+        data_tf = data_tf.reset_joint_velocities(W_ν_plus[6:])
+        data_tf = data_tf.reset_base_velocity(
+            W_ν_plus[0:6], velocity_representation=jaxsim.VelRepr.Inertial
+        )
+
+        return data_tf.replace(
+            velocity_representation=data.velocity_representation, validate=False
+        )
+
+
+@jax.jit
+def step(
+    model: js.model.JaxSimModel,
+    data: js.data.JaxSimModelData,
+    *,
+    dt: jtp.FloatLike,
+    joint_forces: jtp.VectorLike | None = None,
+    link_forces: jtp.MatrixLike | None = None,
+    indices_of_enabled_collidable_points: jtp.VectorLike | None = None,
+) -> tuple[js.data.JaxSimModelData, dict[str, Any]]:
+
+    assert isinstance(model.contact_model, ViscoElasticContacts)
+    visco_elastic_contacts = model.contact_model.replace(
+        parameters=data.contacts_params
+    )
+
```

Comment:
Maybe here we should validate the `indices_of_enabled_collidable_points` argument since after is used directly as array index?

```suggestion

    indices_of_enabled_collidable_points = (
        indices_of_enabled_collidable_points
        if indices_of_enabled_collidable_points is not None
        else jnp.arange(
            len(model.kin_dyn_parameters.contact_parameters.body)
        ).astype(int)
    )
```

Reply:
I refactored the code and removed this extra argument. Now there's a new low-level logic to enable/disable collidable points. Though, it's not yet clear how to expose this to the user. Let's play around with this low-level flag for a while, and then in the future decide if and how expose it.